hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/BatchStageTest.java

```diff
@@ -1035,6 +1035,35 @@ public void hashJoin_when_outputFnReturnsNull_then_filteredOut() {
         assertEquals(emptyList(), new ArrayList<>(sinkList));
     }
 
+    @Test
```

Comment:
This claims to test that `innerHashJoin` filters out nulls, but it actually uses `hashJoin` and asserts that it does not filter out nulls.
About method naming: we use `_` only for special purposes like the structure `when_predondition_then_postcondition`. So this method should be `when_hashJoin_then_dontFilterOutNulls` or similar.